### PR TITLE
GC: If the base snapshot GC data's version doesn't match the current, then ignore it

### PR DESCRIPTION
**Pull request by markfields**
## Description

When we make changes to the GC Data format or semantics, it is difficult to reason over behavior when loading from the old GC Data.

There's no reason to do so anyway since on the next GC run it'll be overwritten from scratch according to the new version.  So don't bother reading it at all.

## Reviewer Guidance

@agarwal-navin - Could/should we remove the logic elsewhere that does fullTree GC when GC Version doesn't match, since it'll happen automatically now?

### CODE REVIEW

File: packages/runtime/container-runtime/src/gc/garbageCollection.ts [hunk 1]
```diff
@@ -216,6 +213,11 @@ export class GarbageCollector implements IGarbageCollector {
 					return undefined;
 				}
 
+				// If the GC Version isn't current, don't load the GC data - we'll regenerate it anyway.
+				if (this.configs.gcVersionInBaseSnapshot !== this.configs.gcVersionInEffect) {
+					return undefined;
+				}
+
```
Comment (markfields) [inline packages/runtime/container-runtime/src/gc/garbageCollection.ts:220]: @agarwal-navin this is the only change, the rest is #14657 (which isn't yet merged).  There's also a new test.

Comment (agarwal-navin) [reply packages/runtime/container-runtime/src/gc/garbageCollection.ts:220]: I was thinking that this check should be done in initializeBaseState and this.initializeGCStateFromBaseSnapshotP but I think this is better. Basically, we don't use this data at all.

File: packages/runtime/container-runtime/src/gc/garbageCollection.ts [hunk 2]
```diff
@@ -295,6 +300,7 @@ export class GarbageCollector implements IGarbageCollector {
 			return { gcData: { gcNodes }, usedRoutes };
 		});
 
+		//* Log current GC Version (maybe add to gcConfigs)
```
Comment (markfields) [inline packages/runtime/container-runtime/src/gc/garbageCollection.ts:303]:
```suggestion
```

**Comment by agarwal-navin** [comment]
> @agarwal-navin - Could/should we remove the logic elsewhere that does fullTree GC when GC Version doesn't match, since it'll happen automatically now?

We should leave that code as-is because its very explicit and clearly states what should happen. It will happen automatically with this PR but this logic can change which will lead to bugs.

**Comment by markfields** [comment]
This was incorporated into a another PR

**Pull request closed without merging**